src/dc_receive_imf.rs

```diff
@@ -110,29 +112,23 @@ pub fn dc_receive_imf(
     // we do not check Return-Path any more as this is unreliable, see
     // https://github.com/deltachat/deltachat-core/issues/150)
     let (from_id, from_id_blocked, incoming_origin) =
-        if let Some(field_from) = mime_parser.get(HeaderDef::From_) {
-            from_field_to_contact_id(context, field_from)?
-        } else {
-            (0, false, Origin::Unknown)
-        };
+        from_field_to_contact_id(context, &mime_parser.from)?;
```

Comment:
See the comment right before `dc_receive_imf`: if message is wrong (has no `From` field?), do not lift up the error, as it will result in fetch retry. Print a warning to log with `warn!` and return `Ok`, indicating that we have completely processed the message and are not going to return to it.